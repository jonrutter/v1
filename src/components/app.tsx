import React from 'react';

import { ColorThemeContextProvider } from '@/hooks/useColorTheme';

/**
 * Wraps all pages on the site.
 *
 * Currently includes the `<ColorThemeContextProvider />`
 */
export const PageWrap: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  return <ColorThemeContextProvider>{children}</ColorThemeContextProvider>;
};
