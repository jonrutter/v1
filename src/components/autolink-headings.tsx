import React, { useEffect, useMemo, useRef, useState } from 'react';
import { Link } from 'gatsby';

// util func
import { formatId } from '@/utils/formatId';

/* Hash Link Component */

// helper hook
const useTitle = (rawTitle: React.ReactNode) => {
  const eleRef = useRef<HTMLHeadingElement>(null);
  const [title, setTitle] = useState<string>('');
  useEffect(() => {
    setTitle(eleRef.current?.textContent || '');
  }, []);
  return useMemo(
    () => [title, eleRef] as [string, React.RefObject<HTMLHeadingElement>],
    [title]
  );
};

const Hash: React.FC = () => (
  <span
    aria-hidden
    className="not-prose ml-2 opacity-0 group-hover:opacity-100 group-focus-visible:opacity-100 select-none pointer-events-none inline-block outline-none"
  >
    #
  </span>
);

/* Header components */
type Props = {
  children: React.ReactNode;
};

export const AutoLinkH2: React.FC<Props> = ({ children }) => {
  const [title, ref] = useTitle(children);
  const id = formatId(title);
  return (
    <h2 ref={ref} id={id} className="not-prose" data-testid={id}>
      <Link
        to={`#${id}`}
        className="not-prose hover:underline focus-visible:underline group outline-none"
      >
        {children}
        <Hash />
      </Link>
    </h2>
  );
};

export const AutoLinkH3: React.FC<Props> = ({ children }) => {
  const [title, ref] = useTitle(children);
  const id = formatId(title);
  return (
    <h3 ref={ref} id={id} className="not-prose group" data-testid={id}>
      <Link
        to={`#${id}`}
        className="not-prose hover:underline focus-visible:underline group outline-none"
      >
        {children}
        <Hash />
      </Link>
    </h3>
  );
};
