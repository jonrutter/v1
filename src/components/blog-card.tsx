import React from 'react';
import { GatsbyImage } from 'gatsby-plugin-image';
import parseISO from 'date-fns/parseISO';
import format from 'date-fns/format';

// types
import type { BlogPostPreview } from '@/types';

// components
import { StyledLink } from './styled-link';

type Props = {
  item: BlogPostPreview;
};

export const BlogCard: React.FC<Props> = ({ item }) => {
  return (
    <article className="shadow-lg rounded-xl overflow-hidden bg-white dark:bg-slate-700 flex flex-col md:flex-row lg:flex-col w-full max-w-[325px] md:max-w-full lg:max-w-[325px] mx-auto text-slate-600 dark:text-slate-200 isolate">
      {/* image wrap */}
      {item.frontmatter.featured_image && (
        <div className="md:flex-[3] lg:flex-auto overflow-hidden">
          <GatsbyImage
            alt={`Preview image for post ${item.frontmatter.title}`}
            image={
              item.frontmatter.featured_image.childImageSharp.gatsbyImageData
            }
            className="max-w-[325px] md:h-full md:object-cover lg:h-auto lg:object-none pointer-events-none select-none"
          />
        </div>
      )}
      {/* content wrap */}
      <div className="max-w-prose p-8 lg:p-6 md:flex-[5] lg:flex-auto">
        <header>
          <h3 className="font-heading font-bold mb-1 text-2xl text-slate-900 dark:text-white">
            <StyledLink to={`/blog/${item.slug}`} className="inline-block">
              {item.frontmatter.title}
            </StyledLink>
          </h3>
          <small className="block mb-4 text-slate-500 dark:text-slate-300">
            <time dateTime={item.frontmatter.date}>
              {format(parseISO(item.frontmatter.date), 'MMMM d, yyyy')}
            </time>{' '}
            â€¢ {item.timeToRead} minute read
          </small>
        </header>
        <div>
          <p className="mb-2">{item.frontmatter.excerpt}</p>
        </div>
      </div>
    </article>
  );
};
