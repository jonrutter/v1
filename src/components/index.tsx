export * from './cards';
export * from './form';
export * from './layout';
export * from './sections';

export * from './app';
export * from './buttons';
export * from './code-block';
export * from './code-tabs';
export * from './hamburger';
export * from './header-links';
export * from './icon';
export * from './portfolio-tabs';
export * from './seo';
export * from './spinner';
export * from './styled-link';
export * from './theme-toggle';
