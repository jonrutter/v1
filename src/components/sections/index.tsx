// export * from './blog-hero';
export * from './cta-box';
export * from './home-hero';
export * from './portfolio-hero';
