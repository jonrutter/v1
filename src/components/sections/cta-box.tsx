import React from 'react';

// components
import { PrimaryButton } from '@/components';

/**
 * Renders a page's CTA Box
 */
export const CTABox: React.FC = () => (
  <div className="px-6 md:px-12 transition-all -mt-20 md:mt-0 bg-inherit text-slate-700 dark:text-slate-200">
    <div className="max-w-site-md lg:max-w-site-lg mx-auto translate-y-1/2 bg-white dark:bg-gradient-to-br dark:from-slate-600 dark:to-slate-800 py-12 px-8 md:px-10 lg:py-16 lg:px-12 rounded-3xl shadow-lg transition-all overflow-hidden">
      <section className="grid grid-cols-1 items-center justify-items-center max-w-sm mx-auto gap-4 md:gap-12 md:grid-cols-2 md:max-w-full lg:grid-cols-3 lg:gap-16">
        <h2 className="transition-all font-heading font-black text-3xl md:text-4xl text-slate-900 dark:text-slate-50">
          Let's get in touch
        </h2>
        <p className="text-lg">
          I'm currently available for work. Get in touch with me for inquiries,
          or just to say hi!
        </p>
        <div className="md:col-span-full lg:col-auto">
          <PrimaryButton to="/contact">Contact me</PrimaryButton>
        </div>
      </section>
      <div className="shine opacity-50 dark:opacity-40" />
    </div>
  </div>
);
