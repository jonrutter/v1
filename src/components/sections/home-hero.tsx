import React from 'react';

// components
import { Hero, PrimaryButton, SecondaryButton } from '@/components';

export const HomeHero: React.FC = () => (
  <Hero title="Jon Rutter" subtitle="Building websites that shine">
    <p className="mb-4">
      Hello and welcome! I'm a front-end web developer specializing in building
      performative, accessible, and responsive websites with React.
    </p>
    <p className="mb-4">
      I'm passionate about building great websites for my clients, contributing
      to open source projects, and expanding my knowledge about web and software
      development.
    </p>
    <p className="mb-4">
      I'm currently available for work. Contact me for inquiries, or just to say
      hello!
    </p>
    <div className="text-center md:text-left mt-8 md:mt-8 flex flex-col sm:flex-row sm:items-center space-y-6 sm:space-y-0 sm:space-x-8">
      <PrimaryButton to="/contact" className="w-full sm:w-auto">
        Contact me
      </PrimaryButton>
      <SecondaryButton to="/portfolio" className="w-full sm:w-auto">
        My portfolio
      </SecondaryButton>
    </div>
  </Hero>
);
