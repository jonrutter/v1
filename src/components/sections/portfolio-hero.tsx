import React from 'react';

// components
import { Hero, PrimaryButton, SecondaryButton } from '@/components';

// social
import { links } from '@/config';

export const PortfolioHero: React.FC = () => (
  <Hero title="My portfolio" subtitle="Check out some of my latest work">
    <p className="mb-4">
      Here are some of my recent projects and client websites.
    </p>
    <p className="mb-4">
      Feel free to also check out my GitHub account to see more of my projects
      and open source contributions.
    </p>
    <div className="text-center md:text-left mt-8 md:mt-8 flex flex-col sm:flex-row sm:items-center space-y-6 sm:space-y-0 sm:space-x-8">
      <PrimaryButton to="/contact" className="w-full sm:w-auto">
        Contact me
      </PrimaryButton>
      <SecondaryButton as="a" href={links.github} className="w-full sm:w-auto">
        My GitHub
      </SecondaryButton>
    </div>
  </Hero>
);
