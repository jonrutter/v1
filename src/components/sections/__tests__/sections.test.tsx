import React from 'react';
import renderer from 'react-test-renderer';

import { CTABox, HomeHero, PortfolioHero } from '@/components/sections';

describe('CTA Box', () => {
  it('renders the correct content', () => {
    const tree = renderer.create(<CTABox />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Home Hero', () => {
  it('renders the correct content', () => {
    const tree = renderer.create(<HomeHero />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Portfolio Hero', () => {
  it('renders the correct content', () => {
    const tree = renderer.create(<PortfolioHero />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
