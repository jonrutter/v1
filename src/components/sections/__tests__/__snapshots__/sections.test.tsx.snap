// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTA Box renders the correct content 1`] = `
<div
  className="px-6 md:px-12 transition-all -mt-20 md:mt-0 bg-inherit text-slate-700 dark:text-slate-200"
>
  <div
    className="max-w-site-md lg:max-w-site-lg mx-auto translate-y-1/2 bg-white dark:bg-gradient-to-br dark:from-slate-600 dark:to-slate-800 py-12 px-8 md:px-10 lg:py-16 lg:px-12 rounded-3xl shadow-lg transition-all overflow-hidden"
  >
    <section
      className="grid grid-cols-1 items-center justify-items-center max-w-sm mx-auto gap-4 md:gap-12 md:grid-cols-2 md:max-w-full lg:grid-cols-3 lg:gap-16"
    >
      <h2
        className="transition-all font-heading font-black text-3xl md:text-4xl text-slate-900 dark:text-slate-50"
        data-testid="cta-box-header"
      >
        Let's get in touch
      </h2>
      <p
        className="text-lg"
      >
        Contact me to discuss anything web development related, or just to say hi!
      </p>
      <div
        className="md:col-span-full lg:col-auto"
      >
        <a
          className="group leading-none py-4 px-8 text-heading font-bold relative inline-block rounded-lg overflow-hidden z-10 transition-all shadow-lg text-lg font-heading bg-slate-900 text-white dark:bg-white dark:text-slate-900 outline-none ring-slate-900 ring-offset-white dark:ring-white dark:ring-offset-slate-900 ring-0 ring-offset-0 focus:ring-2 focus:ring-offset-2 hover:bg-slate-500 dark:hover:bg-slate-400"
          data-testid="cta-box-link"
          href="/contact"
        >
          Contact me
        </a>
      </div>
    </section>
    <div
      className="shine opacity-50 dark:opacity-40"
    />
  </div>
</div>
`;

exports[`Home Hero renders the correct content 1`] = `
<div
  className="py-12 px-6 md:py-24 md:px-12 transition-all bg-white dark:bg-slate-900 text-base sm:text-lg lg:flex lg:items-center lg:min-h-[calc(100vh-5rem)]"
>
  <div
    className="max-w-site-sm md:max-w-site-md lg:max-w-site-lg mx-auto"
  >
    <div
      className="max-w-prose"
    >
      <div
        className="mb-6 md:mb-8 relative z-10 inline-block"
      >
        <h1
          className="transition-all font-heading font-black text-4xl md:text-5xl lg:text-6xl text-slate-900 dark:text-white mb-3 md:mb-4"
        >
          Jon Rutter
        </h1>
        <h2
          className="transition-all font-heading font-black text-2xl md:text-3xl lg:text-4xl text-slate-800 dark:text-slate-100"
        >
          Building websites that shine
        </h2>
        <div
          className="shine opacity-50 dark:opacity-40"
        />
      </div>
      <p
        className="mb-4"
      >
        Hello and welcome! I'm a front-end web developer specializing in building performative, accessible, and responsive websites with React.
      </p>
      <p
        className="mb-4"
      >
        I'm passionate about building great websites for my clients, contributing to open source projects, and expanding my knowledge about web and software development.
      </p>
      <div
        className="text-center md:text-left mt-8 md:mt-8 flex flex-col sm:flex-row sm:items-center space-y-6 sm:space-y-0 sm:space-x-8"
      >
        <a
          className="group leading-none py-4 px-8 text-heading font-bold relative inline-block rounded-lg overflow-hidden z-10 transition-all shadow-lg text-lg font-heading bg-slate-900 text-white dark:bg-white dark:text-slate-900 outline-none ring-slate-900 ring-offset-white dark:ring-white dark:ring-offset-slate-900 ring-0 ring-offset-0 focus:ring-2 focus:ring-offset-2 hover:bg-slate-500 dark:hover:bg-slate-400 w-full sm:w-auto"
          href="/contact"
        >
          Contact me
        </a>
        <a
          className="group leading-none py-[calc(1rem-2px)] px-[calc(2rem-2px)] text-heading font-bold relative inline-block rounded-lg overflow-hidden z-10 transition-all shadow-lg text-lg font-heading bg-white text-slate-500 hover:text-slate-900 focus:text-slate-900 dark:focus:text-white focus:outline-none dark:bg-slate-900 dark:text-slate-300 dark:hover:text-white border-2 border-slate-500 dark:border-slate-300 outline-none ring-slate-900 ring-offset-white dark:ring-white dark:ring-offset-slate-900 ring-0 ring-offset-0 focus:ring-2 focus:ring-offset-2 hover:border-slate-900 dark:hover:border-white w-full sm:w-auto"
          href="/portfolio"
        >
          My portfolio
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Portfolio Hero renders the correct content 1`] = `
<div
  className="py-12 px-6 md:py-24 md:px-12 transition-all bg-white dark:bg-slate-900 text-base sm:text-lg lg:flex lg:items-center lg:min-h-[calc(100vh-5rem)]"
>
  <div
    className="max-w-site-sm md:max-w-site-md lg:max-w-site-lg mx-auto"
  >
    <div
      className="max-w-prose"
    >
      <div
        className="mb-6 md:mb-8 relative z-10 inline-block"
      >
        <h1
          className="transition-all font-heading font-black text-4xl md:text-5xl lg:text-6xl text-slate-900 dark:text-white mb-3 md:mb-4"
        >
          My portfolio
        </h1>
        <h2
          className="transition-all font-heading font-black text-2xl md:text-3xl lg:text-4xl text-slate-800 dark:text-slate-100"
        >
          Check out some of my latest work
        </h2>
        <div
          className="shine opacity-50 dark:opacity-40"
        />
      </div>
      <p
        className="mb-4"
      >
        Here are some of my recent projects and client websites.
      </p>
      <p
        className="mb-4"
      >
        Feel free to also check out my GitHub account to see more of my projects and open source contributions.
      </p>
      <div
        className="text-center md:text-left mt-8 md:mt-8 flex flex-col sm:flex-row sm:items-center space-y-6 sm:space-y-0 sm:space-x-8"
      >
        <a
          className="group leading-none py-4 px-8 text-heading font-bold relative inline-block rounded-lg overflow-hidden z-10 transition-all shadow-lg text-lg font-heading bg-slate-900 text-white dark:bg-white dark:text-slate-900 outline-none ring-slate-900 ring-offset-white dark:ring-white dark:ring-offset-slate-900 ring-0 ring-offset-0 focus:ring-2 focus:ring-offset-2 hover:bg-slate-500 dark:hover:bg-slate-400 w-full sm:w-auto"
          href="/contact"
        >
          Contact me
        </a>
        <a
          className="group leading-none py-[calc(1rem-2px)] px-[calc(2rem-2px)] text-heading font-bold relative inline-block rounded-lg overflow-hidden z-10 transition-all shadow-lg text-lg font-heading bg-white text-slate-500 hover:text-slate-900 focus:text-slate-900 dark:focus:text-white focus:outline-none dark:bg-slate-900 dark:text-slate-300 dark:hover:text-white border-2 border-slate-500 dark:border-slate-300 outline-none ring-slate-900 ring-offset-white dark:ring-white dark:ring-offset-slate-900 ring-0 ring-offset-0 focus:ring-2 focus:ring-offset-2 hover:border-slate-900 dark:hover:border-white w-full sm:w-auto"
          href="https://github.com/rutterjt"
        >
          My GitHub
        </a>
      </div>
    </div>
  </div>
</div>
`;
